WalletWasabi.Tests/UnitTests/Crypto/TranscriptTests.cs

```diff
@@ -0,0 +1,93 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NBitcoin.Secp256k1;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Crypto.Randomness;
+using WalletWasabi.Crypto.ZeroKnowledge;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests.Crypto
+{
+	public class TranscriptTests
+	{
+		[Fact]
+		public void SimpleEquivalenceTest()
+		{
+			var protocol = Encoding.UTF8.GetBytes("test protocol");
+			var realTranscript = new Transcript(protocol);
+			var testTranscript = new TestTranscript(protocol);
+
+			realTranscript.CommitPublicNonces(new[] { Generators.G });
+			testTranscript.CommitPublicNonces(new[] { Generators.G });
+
+			var realChallenge = realTranscript.GenerateChallenge();
+			var testChallenge = testTranscript.GenerateChallenge();
+
+			Assert.Equal(realChallenge, testChallenge);
+		}
+
+		[Fact]
+		public void ComplexEquivalenceTest()
+		{
+			var protocol = Encoding.UTF8.GetBytes("test protocol");
+			var realTranscript = new Transcript(protocol);
+			var testTranscript = new TestTranscript(protocol);
+
+			realTranscript.CommitPublicNonces(new[] { Generators.G });
+			testTranscript.CommitPublicNonces(new[] { Generators.G });
+
+			for (var i = 0; i < 32; i++)
+			{
+				var realChallenge = realTranscript.GenerateChallenge();
+				var testChallenge = testTranscript.GenerateChallenge();
+
+				Assert.Equal(realChallenge, testChallenge);
+
+				realTranscript.CommitPublicNonces(new[] { Generators.G, Generators.GV });
+				testTranscript.CommitPublicNonces(new[] { Generators.G, Generators.GV });
+			}
+		}
+
+		[Fact]
+		public void SyntheticNoncesTest()
```

Comment:
i had some more comprehensive tests in #4203 that should probably be added:

https://github.com/nothingmuch/WalletWasabi/blob/62551f8195480860f096506c2589c98c32f433be/WalletWasabi.Tests/UnitTests/Crypto/ZeroKnowledge/TranscriptTests.cs#L86-L137

Reply:
I cannot add those test right now because they depend on the `Statement` class and the method to commit to a statement is still not there. Lets merge this and once we have the Statement class merged then we can add these tests.

Reply:
oh you're right, i forgot about that